purchase_order_supplier_phone/views/purchase_order_views.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="purchase_order_form_inherit" model="ir.ui.view">
+        <field name="name">purchase.order.form</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_form" />
+        <field name="priority" eval="18" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='partner_id']" position="after">
+                <field name="supplier_phone" />
+                <field name="supplier_mobile" />
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_purchase_order_filter_inherit" model="ir.ui.view">
+        <field name="name">request.quotation.select</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.view_purchase_order_filter" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='name']" position="before">
+                <field
+                    name="supplier_phone"
+                    string="Supplier Phone/Mobile"
+                    filter_domain="['|', ('supplier_phone', 'ilike', self), ('supplier_mobile', 'ilike', self)]"
+                />
+            </xpath>
+
+        </field>
+    </record>
+
+    <record id="view_purchase_order_view_search_inherit" model="ir.ui.view">
+        <field name="name">request.quotation.select</field>
+        <field name="model">purchase.order</field>
+        <field name="inherit_id" ref="purchase.purchase_order_view_search" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='name']" position="before">
+                <field
+                    name="supplier_phone"
+                    string="Supplier Phone/Mobile"
+                    filter_domain="['|', ('supplier_phone', 'ilike', self), ('supplier_mobile', 'ilike', self)]"
+                />
+            </xpath>
+
+        </field>
+    </record>
+
+    <record id="view_purchase_order_tree_inherit" model="ir.ui.view">
+        <field name="name">request.quotation.select</field>
```

Comment:
Do not assign a random name here.  What we normally do is to simply copy the original for the record id and name unless there is a good reason to do otherwise (or people start coming up with random names and could the situation could become untidy).  Please review other records as well.